src/tests/tests/utils.js

```diff
@@ -467,4 +467,56 @@ QUnit.test("getHostFromDomainInput", assert => {
   );
 });
 
+// Tests algorithm used in the pixel tracking heuristic
+// It should return a common substring between two given values
+QUnit.test("findCommonSubstrings", assert => {
+
+  assert.equal(
+    utils.findCommonSubstrings('www.google.com', 'www.google.com/analytics/')[0].length >= 8,
+    true,
+    "returned matches have length greater than or equal to 8"
+  );
+
+  assert.equal(
+    utils.findCommonSubstrings('foobar.com/foo/fizz/buzz/bar', 'foobar.com/foo/bizz/fuzz/bar')[0],
+    'foobar.com/foo/',
+    "returns longest matching value from the pair of URLs"
+  );
+
+  assert.equal(
+    utils.findCommonSubstrings('www.google.com', '.w..wgoogle.analytics.com')[0] === undefined,
+    true,
+    "returns empty array if there is no common substring even in very similar string values"
```

Comment:
Empty array or `undefined`?

Reply:
I decided to test that the first index was undefined as a way of simulating a check that the array would be empty, since if I were to check if two empty arrays were equal to each other it would still run as false. There may be a more graceful way of doing this check

Reply:
`assert.deepEqual` should work ...

Reply:
This is a bit of a nit, but a `deepEqual` comparison between the function call and `[]` is more to the point than comparing the index at 0 of the function call to `undefined`. What if there is something at index 1? We want to know that an empty array was returned.